### PR TITLE
fix: Remove depth and transparentBorders from FieldProps, remove ObjectArrayField, and refactor OpenObjectField

### LINKED ISSUE #6631: [Adaptive-Form] Remove depth and transparentBorders from FieldProps
The Adaptive Form no longer uses transparentBorders and depth can be easy replaced isRoot

### LINKED ISSUE #6632: [Adaptive-Form] Remove ObjectArrayField
Since we are no longer rendering object array field properties inline, we can remove ObjectArrayField and just use the ArrayField.

### LINKED ISSUE #7526: [Adaptive-Form] Refactor the OpenObjectField to use 'Add button' and remove inline fields
## Describe the bug
The `<OpenObjectField />` is the only place in Composer where we use a TextField to add a new item instead of an 'Add' button and is the only place where we still have inline fields. We should update the component to match the rest of Composer's styling.

### LINKED ISSUE #7542: Composer UI does not adequately support action schemas containing for arrays of objects
## Describe the bug
I have a custom action that includes an Array property of type object.

Here is the JSON of what the completed array should look like:
```
[
     {
          "EmailAddress": {
                "Name": "Megan Bowen",
                "Address": "email@domain.com"
           }
     },
     {
          "EmailAddress": {
                "Name": "Joni Sherman",
                "Address": "email@domain.com"
           }
     }
]
```

I wrote the following schema:
```
"attendees": {
      "$ref": "schema:#/definitions/arrayExpression",
      "title": "Attendees",
      "description": "Event attendees",
      "oneOf": [
        {
          "type": "array",
          "title": "Attendees",
          "description": "Event attendees",
          "items": {
            "type": "object",
            "title": "Attendee",
            "description": "Microsoft Graph Attendee",
            "properties": {
              "EmailAddress": {
                "type": "object",
                "title": "Microsoft Graph Email Address",
                "description": "Email address",
                "properties": {
                  "Address": {
                    "type": "string",
                    "title": "Address",
                    "description": "Attendee email address"
                  },
                  "Name": {
                    "type": "string",
                    "title": "Name",
                    "description": "Attendee name"
                  }
                }
              }
            }
          }
        },
        {
          "$ref": "schema:#/definitions/equalsExpression"
        }
      ],
      "examples": [
        "=dialog.event.attendees"
      ]
    }
```

The Composer UI for this property shows like this:
image

The shown UI does not give me any ability to fill in the Address and Name Property, and when I type anything into the box, the underlying JSON creates an array from the entered string and essentially adds garbage to my object.

I would expect to be either just have an Object Editor input for each ArrayItem with the EmailAddress json preconfigured with empty string, or to have a way of entering the Name and Address fields in input boxes.

## Version

Version:  2.0.0-nightly.239644.d95eef8
Electron: 8.2.4
Chrome: 80.0.3987.165
NodeJS: 12.13.0
V8: 8.0.426.27-electron.0

### LINKED ISSUE #7840: Delete properties properties list should not require user to press the Enter key
## Describe the bug

Can enter a property to delete, but it doesn't take unless you also press Enter.

## Version

Version:  2.0.0-nightly.245345.48d2167
Electron: 8.2.4
Chrome: 80.0.3987.165
NodeJS: 12.13.0
V8: 8.0.426.27-electron.0

## Browser

- [x] Electron distribution
- [ ] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

## OS

- [ ] macOS
- [x] Windows
- [ ] Ubuntu

## To Reproduce

1. Add a **Delete properties** action to a dialog.
2. Edit the action's properties.
3. Error: notice that the property doesn't take unless you press Enter.
    - In most other actions, pressing Enter is not required. You only have to add text and then click outside the input box.
    - Even with this action. Once an input box is recognized as having a value, you can change that value and have it stick without having to press Enter first.

## Expected behavior

The semantics of user input boxes is as consistent as possible throughout the product.

## Additional context

- The vertical ellipsis has hover text for the empty input box, but not for an input box that already has a value.

  image

### PR DESCRIPTION
**Pull request by tdurnford**
## Description
- Remove depth and transparentBorders from FieldProps 
- Remove ObjectArrayField
- Update OneOfField dropdown styles
- Updates OpenObjectField to use an 'Add new' button and remove inline fields

## Task Item
Closes #6631
Closes #6632
Closes #7526
Closes #7542
Closes #7840

### CODE REVIEW

File: Composer/packages/adaptive-form/src/components/fields/UnsupportedField.tsx [hunk 1]
```diff
@@ -17,7 +18,7 @@ export const UnsupportedField: React.FC<FieldProps> = function UnsupportedField(
       <div css={unsupportedField.container} data-testid="UnsupportedField">
         {props.label} (Unsupported Field)
         <Link styles={unsupportedField.link} onClick={() => setShowDetails((prev) => !prev)}>
-          Toggle Details
+          {formatMessage('Toggle Details')}
```
Comment (beyackle) [inline Composer/packages/adaptive-form/src/components/fields/UnsupportedField.tsx:21]: 👍

File: Composer/packages/adaptive-form/src/components/fields/OpenObjectField/__tests__/OpenObjectField.test.tsx [hunk 2]
```diff
@@ -72,21 +71,10 @@ describe('<OpenObjectField />', () => {
...
-      expect(onChange).not.toHaveBeenCalled();
-
-      fireEvent.keyDown(getByPlaceholderText('Add a new value'), { key: 'Enter' });
-      expect(onChange).toHaveBeenCalledWith(
-        expect.objectContaining({
-          newKey: 'new value',
-        })
-      );
+      const { findByText } = renderSubject({ schema: { additionalProperties: true }, onChange });
+      await findByText('Add new');
```
Comment (beyackle) [inline Composer/packages/adaptive-form/src/components/fields/OpenObjectField/__tests__/OpenObjectField.test.tsx:77]: Does a rejected Promise cause a unit test to fail, or do we need to wrap this in something?

File: Composer/packages/adaptive-form/src/components/AddButton.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+type Props = {
+  onClick: () => void;
+};
+
+export const AddButton = ({ children, onClick }: React.PropsWithChildren<Props>) => {
+  return (
+    <ButtonContainer>
+      <ActionButton styles={actionButtonStyles} onClick={onClick}>
+        {children ?? formatMessage('Add new')}
```
Comment (beyackle) [inline Composer/packages/adaptive-form/src/components/AddButton.tsx:33]: This feels a little weird taking arbitrary children; shouldn't this be a string prop (with the same default)?

**Review by beyackle — CHANGES_REQUESTED** [review]

File: Composer/packages/adaptive-form/src/components/fields/OpenObjectField/__tests__/OpenObjectField.test.tsx (hunk 2 above)
Comment (tdurnford) [reply Composer/packages/adaptive-form/src/components/fields/OpenObjectField/__tests__/OpenObjectField.test.tsx:77]: Yes, a rejected promise will cause a test to fail

File: Composer/packages/adaptive-form/src/components/AddButton.tsx (hunk 3 above)
Comment (tdurnford) [reply Composer/packages/adaptive-form/src/components/AddButton.tsx:33]: I wanted the component to behave exactly like the `<ActionButton />` which I believe takes arbitrary children.

**Review by beyackle — APPROVED** [review]
I'm satisfied with the answers I got, so ✅

**Pull request merged**